lib/faraday/connection.rb

```diff
@@ -464,5 +442,14 @@ def set_authorization_header(header_type, *args)
         header(*args)
       headers[Faraday::Request::Authorization::KEY] = header
     end
+
+    def find_default_proxy
+      warn 'no_proxy is unsupported' if ENV['no_proxy'] || ENV['NO_PROXY']
```

Comment:
You mean when you initialise the connection simply as `conn = Faraday.new` ?
That is actually true, I missed this detail as it's an unusual way to use it.
This could be easily fixed moving the #find_proxy call on the `run_request` or `build_request` methods.